src/java.base/share/classes/java/io/RandomAccessFile.java

```diff
@@ -75,6 +75,8 @@ public class RandomAccessFile implements DataOutput, DataInput, Closeable {
 
     private volatile boolean closed;
 
+    private final byte[] readBuffer = new byte[8];
```

Comment:
As `readBuffer` is used in multiple places without synchronization, doesn't this create a race condition?

Reply:
It does in case one deliberate accesses the same instance from multiple threads, but there are no thread-safety guarantees for RandomAccessFile. The same shared buffer approach is used in `DataInputStream` so I think we can use it here as well.